### DISCUSSION TITLE
Enforcing specific cApItaLiZaTioN rules

### DISCUSSION THREAD

**Discussion by etdr** [beetbox/beets#4921] [Q&A]
tl;dr what's the best way to make sure that specific capitalization rules are enforced instead of Just Title Caps For Every Word Including Small Words Like With, And, And The?

I'm looking at the [inline plugin](https://beets.readthedocs.io/en/stable/plugins/inline.html) but I also need it for tagging. I guess this isn't the MusicBrainz way probably but hey it ain't proper English (or, really, it's not my style :sunglasses:).

Examples (making these up):
- "Walking With The Dog" → "Walking with the Dog" (words on a defined blocklist of articles and prepositions aren't capitalized by default)
- "the angry dinosaur" → "The Angry Dinosaur" (always capitalize first word, maybe also last word?)
- "Whatever/The Maelstrom" → no change (in certain contexts, e.g. after slashes or if it's the first word it an parenthetical, keep all words capitalized)
- "sUpErFLuiDiTy" → ask me? (could be a bad tag or intentional)

I am capable of writing the Python to catch all this so mainly I'm wondering if there's an existing customizable plugin I'm missing or other existing functionality in beets that would take care of this.

Related: does anyone have an English styleguide lying around?

**Comment by wisp3rwind** [comment]
If I'm not missing anything, the inline plugin should in fact enable you do code this up in Python when combined with [formatted modify](https://github.com/beetbox/beets/pull/4095).

The interactive use case ("sUpErFLuiDiTy" → ask me?) seems harder to achieve, but maybe the confirmation prompt in the `modify` command is good enough?
